### DISCUSSION TITLE
当topic/group超过3w后,broker无法注册到namesrv

### DISCUSSION THREAD

**Discussion by xiaoma20082008** [apache/rocketmq#8107] [Q&A] [answered]
需要调整参数：`com.rocketmq.remoting.frameMaxLength`

**Comment by RongtongJin** [comment, marked as answer]
或者打开enableSplitRegistration

**Comment by xiaoma20082008** [comment]
能不能上报总的topic个数指标和总的group个数的指标，这样可以添加告警

**Reply by leizhiyuan** [reply]
这个功能，可以在exporter中实现
https://github.com/apache/rocketmq-exporter

**Reply by RongtongJin** [reply]
也可以在rocketmq内核的metrics中增加

**Reply by xiaoma20082008** [reply]
> 这个功能，可以在exporter中实现 https://github.com/apache/rocketmq-exporter

也是可以的，但是exporter里需要查询所有的消费组，然后统计数量，在group多的时候，也容易超时和报错。如果直接加metrics，可能更方便一些
